packages/framework-provider-aws/src/library/searcher-adapter.ts

```diff
@@ -32,62 +31,150 @@ export async function searchReadModel(
+        case 'and':
+        case 'or':
+          return `(${(filter as Array<FilterFor<any>>)
+            .map((arrayFilter) => buildFilterExpression(arrayFilter))
+            .join(` ${propName} `)})`
+        default:
+          return buildOperation(propName, filter)
+      }
+    })
     .join(' AND ')
 }
 
-function buildOperation(propName: string, filter: Filter<any>): string {
+function buildOperation(propName: string, filter: Operation<any> = {}): string {
   const holder = placeholderBuilderFor(propName)
-  switch (filter.operation) {
-    case '=':
-      return `#${propName} = ${holder(0)}`
-    case '!=':
-      return `#${propName} <> ${holder(0)}`
-    case '<':
-      return `#${propName} < ${holder(0)}`
-    case '>':
-      return `#${propName} > ${holder(0)}`
-    case '>=':
-      return `#${propName} >= ${holder(0)}`
-    case '<=':
-      return `#${propName} <= ${holder(0)}`
-    case 'in':
-      return `#${propName} IN (${filter.values.map((value, index) => holder(index)).join(',')})`
-    case 'between':
-      return `#${propName} BETWEEN ${holder(0)} AND ${holder(1)}`
-    case 'contains':
-      return `contains(#${propName}, ${holder(0)})`
-    case 'not-contains':
-      return `NOT contains(#${propName}, ${holder(0)})`
-    case 'begins-with':
-      return `begins_with(#${propName}, ${holder(0)})`
-    default:
-      throw new InvalidParameterError(`Operator "${filter.operation}" is not supported`)
-  }
+  return Object.entries(filter)
+    .map(([operation, value], index) => {
+      switch (operation) {
+        case 'eq':
+          return `#${propName} = ${holder(index)}`
+        case 'ne':
+          return `#${propName} <> ${holder(index)}`
+        case 'lt':
+          return `#${propName} < ${holder(index)}`
+        case 'gt':
+          return `#${propName} > ${holder(index)}`
+        case 'gte':
+          return `#${propName} >= ${holder(index)}`
+        case 'lte':
+          return `#${propName} <= ${holder(index)}`
+        case 'in':
+          return `#${propName} IN (${value
+            .map((value: any, subIndex: number) => `${holder(index, subIndex)}`)
+            .join(',')})`
+        case 'contains':
+          return `contains(#${propName}, ${holder(index)})`
+        case 'beginsWith':
+          return `begins_with(#${propName}, ${holder(index)})`
+        case 'includes':
+          return `contains(#${propName}, ${holder(index)})`
+        default:
+          if (typeof value === 'object') {
+            return `#${propName}.${buildOperation(operation, value)}`
+          }
+          throw new InvalidParameterError(`Operator "${operation}" is not supported`)
+      }
+    })
+    .join(' AND ')
 }
 
-function placeholderBuilderFor(propName: string): (valueIndex: number) => string {
-  return (valueIndex: number) => `:${propName}_${valueIndex}`
+const attrCount: string[] = []
+function resetAttrCount(): void {
+  attrCount.length = 0
+}
```

Comment:
**Related to the proposal to delete the global `attrCount` array:**

This code would not be needed anymore